spec/presenters/step_by_step_page_presenter_spec.rb

```diff
@@ -40,4 +39,46 @@
       expect(subject).to eq(summary_with_links_checked)
     end
   end
+
+  describe "#broken_link_status_text" do
+    let(:step) { create(:step) }
+    let(:step_nav) { create(:step_by_step_page, slug: 'step-by-step-with-one-step', steps: [step]) }
+    let(:broken_link) {
+      {
+        "uri": "https://www.gov.uk/404",
+        "status": "broken",
+        "checked": "2017-04-12T16:30:39Z",
+        "errors": [
+          "Received 404 response from the server."
+        ],
+        "warnings": [],
+        "problem_summary": "404 error (page not found)",
+        "suggested_fix": ""
+      }
+    }
+
+    subject { described_class.new(step_nav) }
+
+    before do
+      allow(step).to receive(:link_report?).and_return(true)
+    end
+
+    it "should say 'No broken links' if there are no broken links" do
+      allow(step).to receive(:broken_links?).and_return(false)
+
+      expect(subject.broken_link_status_text(step)).to eq "No broken links"
+    end
+
+    it "should use singular 'link' if only one link is broken" do
+      allow(step).to receive(:broken_links).and_return([broken_link])
+
+      expect(subject.broken_link_status_text(step)).to eq "1 broken link found"
+    end
+
+    it "should pluralise 'links' if multiple links are broken" do
+      allow(step).to receive(:broken_links).and_return([broken_link, broken_link])
```

Comment:
The content of the broken links doesn't really matter for this test, so is `broken_link` needed, or can `broken_links` just return strings?